Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/ObjectIDDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,178 @@
+{
+  [Test]
+  public void GetParameterValue ()
+  {
+    var storageTypeInformationMock = new Mock<IStorageTypeInformation>();
+    var dummyValue = DomainObjectIDs.Order1;
+
+    var dummyGuidResult = Guid.NewGuid();
+
+    storageTypeInformationMock
+        .Setup(_ => _.ConvertToStorageType(dummyValue.Value))
+        .Returns(dummyGuidResult);
+    storageTypeInformationMock.Setup(_ => _.DotNetType).Returns(typeof(Guid));
+
+    var objectIDDataParameterDefinition = new ObjectIDDataParameterDefinition(storageTypeInformationMock.Object);
+
+    var result = objectIDDataParameterDefinition.GetParameterValue(dummyValue);
+
+    Assert.That(result, Is.EqualTo(dummyGuidResult));
+    storageTypeInformationMock.Verify();
+  }
+
+  [Test]
+  public void GetParameterValue_WithNull ()
+  {
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+    object dummyValue = null;
+
+    storageTypeInformationStub
+        .Setup(_ => _.ConvertToStorageType(dummyValue))
+        .Returns(DBNull.Value);
+    storageTypeInformationStub.Setup(_ => _.DotNetType).Returns(typeof(Guid));
+
+    var objectIDDataParameterDefinition = new ObjectIDDataParameterDefinition(storageTypeInformationStub.Object);
+
+    var result = objectIDDataParameterDefinition.GetParameterValue(dummyValue);
+
+    Assert.That(result, Is.EqualTo(DBNull.Value));
+  }
+
+  [Test]
+  public void GetParameterValue_WithInvalidValue_ThrowsArgumentException ()
+  {
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+    var dummyValue = new ObjectID(typeof(Customer), Guid.NewGuid());
+
+    storageTypeInformationStub.Setup(_ => _.DotNetType).Returns(typeof(int));
+
+    var objectIDDataParameterDefinition = new ObjectIDDataParameterDefinition(storageTypeInformationStub.Object);
+
+    Assert.That(
+        () => objectIDDataParameterDefinition.GetParameterValue(dummyValue),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            string.Format("Parameter '{0}' has type '{2}' when type '{1}' was expected.", "parameterObject.Value", typeof(int), typeof(Guid)),
+            "parameterObject.Value"));
+  }
+
+
+  [Test]
+  public void CreateDataParameter_WithIntValue ()
+  {
+    var parameterValue = 42;
+    var commandMock = new Mock<IDbCommand>(MockBehavior.Strict);
+    var dataParameterMock = new Mock<IDbDataParameter>(MockBehavior.Strict);
+
+    var storageTypeInformation = new StorageTypeInformation(
+        typeof(int),
+        "test",
+        DbType.Int32,
+        false,
+        null,
+        typeof(int),
+        new DefaultConverter(typeof(int)));
+
+    commandMock.Setup(mock => mock.CreateParameter()).Returns(dataParameterMock.Object).Verifiable();
+
+    dataParameterMock.SetupSet(mock => mock.DbType = storageTypeInformation.StorageDbType).Verifiable();
+    dataParameterMock.SetupSet(mock => mock.Value = parameterValue).Verifiable();
```

Comment:
I like to use linebreaks for more complicated (or longer) mock setups. Inparticular also verifiables to make it easier discoverable.